### PR TITLE
add get_backfills_count

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
Adds `get_backfills_count` method to return the number of backfills that match a given filter

corresponding internal pr https://github.com/dagster-io/internal/pull/11795

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by jamiedemaria** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24755?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24755** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24755?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24754** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24754?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24745** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24745?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jamiedemaria and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py [hunk 1]
```diff
@@ -948,6 +948,28 @@ def get_backfills(
             ]
         return backfill_candidates
 
+    def get_backfills_count(self, filters: Optional[BulkActionsFilter] = None) -> int:
+        subquery = db_subquery(self._backfills_query(filters=filters))
+        if filters.tags:
+            # runs can have more tags than the backfill that launched them. Since we filtered tags by
+            # querying for runs with those tags, we need to do an additional check that the backfills
+            # also have the requested tags. This requires fetching the backfills from the db and filtering them
+            rows = self.fetchall(subquery)
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py:957]: I'd prefer to not have to fetch and deserialize all of the matching backfills just to check that they actually match the tags we are filtering on. But I'm not sure how else to do this

File: python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py [hunk 2]
```diff
@@ -955,6 +955,27 @@ def get_backfills(
             )
         return backfill_candidates
 
+    def get_backfills_count(self, filters: Optional[BulkActionsFilter] = None) -> int:
+        check.opt_inst_param(filters, "filters", BulkActionsFilter)
+        if filters and filters.tags:
+            # runs can have more tags than the backfill that launched them. Since we filtered tags by
+            # querying for runs with those tags, we need to do an additional check that the backfills
+            # also have the requested tags. This requires fetching the backfills from the db and filtering them
+            query = self._backfills_query(filters=filters)
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py:964]: this is a bit of a bummer that i'm fetching the actual backfill objects from the db, but i'm not sure how else to check that the tags are actually on the backfill object and not just the runs that are launched

Comment (prha) [reply python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py:964]: Can you remind me: how do backfill runs have more tags than the backfills themselves?

Comment (jamiedemaria) [reply python_modules/dagster/dagster/_core/storage/runs/sql_run_storage.py:964]: if you have something like 
```python 
@asset(tags={"foo": "bar"})
...
```
 then a for a backfill of the asset, all of the runs will have the `foo:bar` tag but the backfill object itself won't have that tag

**Review by prha — APPROVED** [review]

**Pull request merged**